api/types/templates.go

```diff
@@ -42,3 +42,30 @@ type TemplateServer struct {
 
 // TemplateScriptCredentials stores credentials to servers
 type TemplateScriptCredentials interface{}
+
+// cookbookVersionsMap stores template cookbook versions map
+type cookbookVersionsMap map[string]cookbookVersionsFields
+
+// cookbookVersionsFields stores cookbook versions fields: version/version_id
+type cookbookVersionsFields struct {
+	Version          string `json:"version,omitempty" header:"VERSION" show:"nolist,noshow"`
+	VersionId        string `json:"version_id,omitempty" header:"VERSION_ID" show:"nolist,noshow"`
+	VersionComposite string `json:"omitempty" header:"VERSION_COMPOSITE" show:"nolist,noheader"`
+}
+
+// FillInCookbookVersion resolves adequate cookbook version from version_id
+func (cbf *Template) FillInCookbookVersion(VersionByVersionID map[string]string) {
```

Comment:
cbf is not a good name for a template, the parameter should start with lowercase and the method should be named FillInCookbookVersionComposite or similar

Reply:
Done!